libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -897,6 +899,71 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
+{
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				if gt(newLen, <maxArrayLength>) {
+					<panic>()
+				}
+
+				let oldLen := <fetchLength>(array)
+
+				if gt(newLen, oldLen) {
+					sstore(array, or(shl(newLen, 1), iszero(lt(newLen, 32))))
```

Comment:
Can you explain this line? Why don't you need an sload here?

Reply:
Ah, this is resizing to bigger array...
Will need to fix this with proper copying from small to bigger array (from packed together with length to unpack)